sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -180,9 +180,18 @@ def _buildprovider(
     # pylint:disable=protected-access
     provider = AzureAppConfigurationProvider()
     headers = kwargs.pop("headers", {})
-    headers["Correlation-Context"] = _get_correlation_context(key_vault_options)
+
+    if os.environ.get(RequestTracingDisabledEnvironmentVariable, default="").lower() != "true":
+        headers["Correlation-Context"] = _get_correlation_context(key_vault_options)
+
     useragent = USER_AGENT
 
+    if "retry_total" not in kwargs:
+        kwargs["retry_total"] = 2
```

Comment:
retry_total = kwargs.pop("retry_total", 2)
then use retry_total=retry_total instead of 
kwargs["retry_total"] = 2